### ISSUE TITLE
How to redirect all traffic to only ONE host containter?

### ISSUE THREAD

**Issue by FractalMind** [JrCs/docker-letsencrypt-nginx-proxy-companion#315]
Excellent job guys! I'm very impressed with your work!

### Where in the code I could modify the internal router that route the SNI "website1.com" TO container "website1" created with VIRTUAL_HOST=website1.com... So it can redirect all traffic to the SAME container, say: "localhost"???
EX:
exemple.com => host container:"localhost"
mycustomdomain.com => host container:"localhost"
anothercustomdomain.com =>host container:"localhost"
ALL 3 HOST CONTAINER ARE THE SAME!

We're developing an app like "_wix.com, squarespace.com_" to generate (maybe millions of) websites on the fly with HTTPS support. This docker project is the closest I've found to meet our criteria to solve our TLS termination problem:(How to host thousands of custom domains with SSL).

The only dilemma is this project was designed for websites with different code
EX:
website1.com => custom code for website1
website2.com => custom code for website2
Each websites are separated with their own code; which is logical.

The problem is our app is already polymorphic(will pull the custom data according to the hostname)
EX:
website1.com => uses the SAME code to build website1 with DIFFERENT data according to the hostname.
website2.com => uses the SAME code to build website2 with DIFFERENT data according to the hostname.

Long story short: We cannot have 1,000,000 containers: website1,website2,...website100001; it makes no sense. It's all the same code! We just need "on the fly TLS termination". I hope to decrypt/generate all SSL on the fly and bump all the traffic TO ONE CONTAINER for our code to do the rest of the magic.

How can I modify the "**docker-letsencrypt-nginx-proxy-companion**" system to only have one host container so I can add support for an unlimited number of custom domain?

Thank you for your time :)

**Comment by buchdag** [comment]
This project (`JrCs/docker-letsencrypt-nginx-proxy-companion`) "only" purpose is to:

-  enable the automatic creation of certificates from containers env var.
-  put them in the right place with the right name so they can be used by nginx-proxy docker-gen template.
-  reload the nginx proxy when needed (cert creation or renewal).

What you want to achieve depends entirely on the [`nginx-proxy`](https://github.com/jwilder/nginx-proxy) project, which does the actual proxying and is maintained by different people.

If your plan is to generate millions of certificates on the fly, you should also be aware of Let's Encrypt [rate limits](https://letsencrypt.org/docs/rate-limits/). Best case scenario is 2000 unique subdomains per week, meaning that getting certificates for a million different domains / subdomains through LE will take a bit less than ten years.

**Comment by buchdag** [comment]
@cpu, is there a rate limit for registered domains per account per week ? I mean would boulder issue certificates for one million discrete registered domains without hitting any rate limit other than requests per second ?

**Comment by cpu** [comment]
@buchdag If they are distinct domain names it wouldn't be a problem provided the overall requests rate limit was respected.

**Comment by FractalMind** [comment]
Thank you for your quick answer guys!

But I still have to find a solution to this problem.

@evertramos provided a [solution](https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion/issues/28#issuecomment-354576104)
Tl;dr: You can add n host into VIRTUAL_HOST=website1.com,website2.com,...website10000.com

But as explained, I would like to not use this method. Somewhere in the code you probably have something like.
redirectToVHost $hostname
Where the proxy forwards everything to the "$hostname container". 
_Ex: exemple.com => VIRTUAL_HOST=exemple.com_

## **Can any of you point me to the script or configuration file where this redirect is made?**

I'll share, of course, my solution once found
Thank you for your time :)

**Comment by FractalMind** [comment]
I think I'm getting close. I've found in /nginx.tmpl
line 122:
`{{ $upstream_name := "testtest.com" }}`

line 269:
```
server {
    [...]
    location / {
        [...]
        include uwsgi_params;
        uwsgi_pass {{ trim $proto }}://{{ trim $upstream_name }};
    }
}
```
But I get a: 
`2018/01/19 03:15:32 [emerg] 1#1: invalid host in upstream "http://thisismysupertest.com" in /etc/nginx/conf.d/default.conf:80`
or
`2018/01/19 03:15:32 [emerg] 1#1: invalid host in upstream "http://172.18.0.3" in /etc/nginx/conf.d/default.conf:80`
172.18.0.3 is the ip of thisismysupertest.com container

Any idea?

**Comment by buchdag** [comment]
Please read https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/issues/315#issuecomment-356681055 again.

This is not nginx-proxy repo.

**Comment by evertramos** [comment]
@FractalMind you should be posting this on https://github.com/jwilder/nginx-proxy

**Comment by FractalMind** [comment]
Oups! Wrong forum haha
Thanks again

**Comment by FractalMind** [comment]
Leaving a breadcrumbs trail for people with the same requirements
[nginx-proxy](https://github.com/jwilder/nginx-proxy/issues/1035)
